common/v2/features/Dashboard/Settings/Import/components/ImportBox.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import styled from 'styled-components';
+import translate from 'translations';
+
+const FilePicker = styled.label`
+  background: none;
+  color: #1eb8e7;
+  cursor: pointer;
+`;
+const FilePickerInput = styled.input`
+  display: none !important;
+`;
+
+interface ImportBoxContainerProps {
+  dragging: boolean;
+}
+
+const ImportBoxContainer =
+  styled.div <
+  ImportBoxContainerProps >
+  `
+  color: #9b9b9b;
+  background: #e8eaed;
+  padding: 6rem;
+  border-radius: 0.375em;
+  box-shadow: ${props => (props.dragging ? '0px 0px 0px 2px #1eb8e7;' : 'none')};
+`;
+
+const ErrorMessage = styled.span`
+  color: #ef4747;
+`;
+
+interface ImportProps {
+  importCache(importedCache: any): void;
+  onNext(): void;
+}
+
+export default class ImportBox extends React.Component<ImportProps> {
+  public state = { badImport: true, dragging: false };
+  public submit = (importedCache: string) => {
+    const importSuccess = this.props.importCache(importedCache);
+    if (Boolean(importSuccess) === false) {
+      this.setState({ badImport: true });
+    }
+    if (Boolean(importSuccess) === true) {
+      this.props.onNext();
+    }
+  };
+
+  public render() {
+    const { badImport, dragging } = this.state;
+    return (
+      <ImportBoxContainer
+        onDrop={this.handleFileSelection}
+        onDragEnter={() => this.setState({ dragging: true })}
+        onDragLeave={() => this.setState({ dragging: false })}
+        dragging={dragging}
+      >
+        {badImport ? (
+          <ErrorMessage>{translate('SETTINGS_IMPORT_INVALID')}</ErrorMessage>
+        ) : (
+          translate('SETTINGS_IMPORT_COPY')
+        )}
+        <br />
+        <br />
+        <FilePicker htmlFor="upload">
+          {translate('SETTINGS_IMPORT_BUTTON')}
+          <FilePickerInput id="upload" type="file" onChange={this.handleFileSelection} />
+        </FilePicker>{' '}
+        {translate('SETTINGS_IMPORT_PASTE')}
+        <br />
+      </ImportBoxContainer>
+    );
+  }
+
+  // private checkPastedCache = (e: any) => {
+  //   if (isValidCache(this.props.localStorage, e.target.value)) {
+  //     this.setState({ isValid: true });
+  //   } else {
+  //     this.setState({ badImport: true });
+  //   }
+  // };
+
+  private handleFileSelection = (e: any) => {
+    const fileReader = new FileReader();
+
+    fileReader.onload = () => {
+      if (fileReader.result) {
+        // this.setState({ isValid: true, importedCache: fileReader.result });
+        this.submit(fileReader.result as string);
+      }
+    };
+    if (e.target.files) {
```

Comment:
nested conditionals should be avoided as much as possible ([holding a program in ones head](http://www.paulgraham.com/head.html))

for the mean time the quick fix is AND
```
if (e.target.files && e.target.files[0])
```